src/dataframerow/dataframerow.jl

```diff
@@ -272,6 +272,12 @@ function Base.names(r::DataFrameRow, cols)
     return [string(nms[i]) for i in idxs]
 end
 
+Base.names(r::DataFrameRow, T::Type) =
+    [String(n) for n in _names(index(r)) if eltype(parent(r)[!, n]) <: T]
```

Comment:
Why not this (given that the method is defined just below)?

```suggestion
    [n for n in _names(r) if eltype(parent(r)[!, n]) <: T]
```
Though this is a case where `pairs` could be a bit simpler.

Reply:
again - we need to return `String`s not `Symbol`s here. I will update it accordingly.